hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -137,6 +138,14 @@ protected HoodieBackedTableMetadataWriter(Configuration hadoopConf, HoodieWriteC
   private HoodieWriteConfig createMetadataWriteConfig(HoodieWriteConfig writeConfig) {
     int parallelism = writeConfig.getMetadataInsertParallelism();
 
+    // The metadata table timeline should always encompass the dataset timeline so that syncing of manual rollbacks can
+    // always check whether the rolled-back-instant was ever synced to the metadata table. Since all actions on the
+    // dataset always lead to a delta-commit on the Metadata Table, we need to preserve enough deltacommits that can
+    // cover total number of all the non-archived actions on the dataset.
+    int multiplier = HoodieTimeline.VALID_ACTIONS_IN_TIMELINE.length;
```

Comment:
why is this multiplier chosen? It's not guaranteed that the timeline will have all valid actions at once right?

Reply:
I wonder if say 8x-ing the retention of the metadata timeline is really necessary.

Reply:
Remove this as this was leading to too many instants retained on metadata table.